### DISCUSSION TITLE
Empty Data Fields

### DISCUSSION THREAD

**Discussion by mzm-s** [theOehrly/Fast-F1#123] [General]
I've noticed this before too but is there a reason why there seems to be empty data points in say for example lap-by-lap Speed Trap data. Apart from the obvious pitstops and such I've noticed SpeedST has a lot more empty fields compared to SpeedFL (when comparing over all 50+ laps in a GP). Any reason why this is? Maybe the way the data is acquired allows for some invalid entries?
eg: BahrainGP 2022: LEC Laps ( 7, 10 to 13, 15, etc.)

**Comment by theOehrly** [comment]
I think the data  really might be missing. But I'll take another look at it, now that you're asking.

**Reply by mzm-s** [reply]
Thanks for taking a look. I know its not the first time I've seen this too. 
At least in the recent GP it was the speed trap data. Odd how some particular laps turn up empty even though others are fine. I cant see why they would be any different in terms of the retrieving process either.

**Reply by theOehrly** [reply]
So, it did check again and the values really are missing. I don't have an explanation for it. But the data is just not there,

**Reply by mzm-s** [reply]
Did a cursory look too at the Saudi Arabian GP. Looks like SpeedTrap, FinishLine and Intermediate #2 Speeds are recorded consistently even though I see a couple of blanks in Intermediate #1. Being that the ST and FL are the most prominent comparison this seems fine to me. Still there's something funky going on being that I1 still is missing some data. Lets see what turns up in Australia.

**Reply by theOehrly** [reply]
When I took a look at the raw api response for individual drivers, the 'I2' key didn't even exist as often, as the number of laps driven by that driver. So there couldn't possibly be a value for every lap. Additionally, not every occurrence of that key actually contained a valid top speed value.

Of course, if you have the time for it, feel free to dig deeper and see what you can find. Maybe I am wrong.

**Reply by mzm-s** [reply]
Sounds good. I've only primarily looked at ST and FL but I'll keep an eye out see if there's something of a pattern to this. Thanks

**Reply by theOehrly** [reply]
As far as I can see, the official app only shows one top speed reading. So why are there four in the data even? It's not quite easily understandable.
